src/classic/element/__tests__/ReactElementValidator-test.js

```diff
@@ -362,6 +362,32 @@ describe('ReactElementValidator', function() {
     expect(console.warn.calls.length).toBe(2);
   });
 
+  it('should warn if a PropType creator is used as a PropType', function() {
+    spyOn(console, 'warn');
+
+    var Component = React.createClass({
+      propTypes: {
+        prop: React.PropTypes.shape
+      },
+      render: function() {
+        return React.createElement('span', null, this.props.prop.value);
+      }
+    });
+
+    ReactTestUtils.renderIntoDocument(
+      React.createElement(Component, {prop: {value: 'hi'}})
+    );
+
+    expect(console.warn.calls.length).toBe(1);
+    expect(console.warn.calls[0].args[0]).toBe(
+      'Warning: Component: the type of prop `prop` is invalid; the type ' +
```

Comment:
nit: call the prop something other than `prop` so we can distinguish these two arguments?